modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/folders/SuggestedFoldersPaywallBottomSheet.kt

```diff
@@ -0,0 +1,64 @@
+package au.com.shiftyjelly.pocketcasts.podcasts.view.folders
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.core.view.doOnLayout
+import androidx.fragment.compose.content
+import au.com.shiftyjelly.pocketcasts.compose.AppTheme
+import au.com.shiftyjelly.pocketcasts.settings.onboarding.OnboardingFlow
+import au.com.shiftyjelly.pocketcasts.settings.onboarding.OnboardingLauncher
+import au.com.shiftyjelly.pocketcasts.settings.onboarding.OnboardingUpgradeSource
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
+import au.com.shiftyjelly.pocketcasts.views.extensions.setSystemWindowInsetToPadding
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class SuggestedFoldersPaywallBottomSheet : BottomSheetDialogFragment() {
+
+    @Inject
+    lateinit var theme: Theme
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ) = content {
+        AppTheme(theme.activeTheme) {
+            SuggestedFoldersPaywall(
+                onUseTheseFolders = {
+                    dismiss()
+                    val source = OnboardingUpgradeSource.PROFILE
+                    val onboardingFlow = OnboardingFlow.PlusAccountUpgrade(source)
+                    OnboardingLauncher.openOnboardingFlow(activity, onboardingFlow)
+                },
+                onMaybeLater = {
+                    dismiss()
+                },
+            )
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        view.doOnLayout {
+            val dialog = dialog as BottomSheetDialog
+            val bottomSheet = dialog.findViewById<View>(com.google.android.material.R.id.design_bottom_sheet) as FrameLayout?
+            if (bottomSheet != null) {
+                BottomSheetBehavior.from(bottomSheet).run {
+                    state = BottomSheetBehavior.STATE_EXPANDED
+                    peekHeight = 0
+                    skipCollapsed = true
+                }
+            }
+        }
+        view.setSystemWindowInsetToPadding(right = true, left = true)
```

Comment:
There are [Compose APIs](https://developer.android.com/reference/kotlin/androidx/compose/foundation/layout/WindowInsets) for handling insets.